0xStation/station-terminal
# Add ethers.js and hydrate front-page with data from blockchain

## Changes

- Make the home page route `/<dao-name>` rather than `/`. 
- Add ethers.js
- Add ABIs for each Default OS contract.
- On home page load, query for the module addresses of that OS, build the ethers.js contracts for each module, and save these contracts to the React context.
- Create a context for the account address. Automatically update the account address when account is changed on Metamask. Display current account address in navbar.
- Query blockchain to hydrate home page with data. **Note:** Some of the fields are still hardcoded because they're going to be hydrated with subgraph data. 
- Move several large components out of index.js page into their own files to improve readability.

## How to test
- Follow the local dev instructions in the updated readme
- Confirm that the account address changes when you change your address in Metamask
- http://localhost:3000 should be empty aside from the navbar/background. http://localhost:3000/Default%20Dao should hydrate with data from your local blockchain
